subsys/bluetooth/services/dis.c

```diff
@@ -101,6 +138,74 @@ static ssize_t read_pnp_id(struct bt_conn *conn,
 }
 #endif
 
+#if CONFIG_BT_DIS_SYSTEM_ID
+static ssize_t read_system_id(struct bt_conn *conn,
+			   const struct bt_gatt_attr *attr, void *buf,
+			   uint16_t len, uint16_t offset)
+{
+	return bt_gatt_attr_read(conn, attr, buf, len, offset, dis_system_id,
+				 sizeof(dis_system_id));
+}
+#endif
+
+#if CONFIG_BT_DIS_UDI
+#define BT_DIS_UDI_FLAG_LABEL		(!!BT_DIS_UDI_LABEL_STR_REF[0])
+#define BT_DIS_UDI_FLAG_DI			(!!BT_DIS_UDI_DI_STR_REF[0])
+#define BT_DIS_UDI_FLAG_ISSUER		(!!BT_DIS_UDI_ISSUER_STR_REF[0])
+#define BT_DIS_UDI_FLAG_AUTHORITY	(!!BT_DIS_UDI_AUTHORITY_STR_REF[0])
+#define BT_DIS_UDI_FLAGS			(BT_DIS_UDI_FLAG_LABEL | (BT_DIS_UDI_FLAG_DI << 1) | \
+	(BT_DIS_UDI_FLAG_ISSUER << 2) | (BT_DIS_UDI_FLAG_AUTHORITY << 3))
+
+#if defined(CONFIG_BT_SETTINGS)
+static uint8_t bt_dis_udi_merged[MIN(CONFIG_BT_DIS_STR_MAX * 4 + 1, BT_ATT_MAX_ATTRIBUTE_LEN)];
```

Comment:
Suggest to add a comment why it needs to store 4 x CONFIG_BT_DIS_STR_MAX

Reply:
Added